src/settings/mod.rs

```diff
@@ -238,27 +239,97 @@ impl From<&Configuration> for RuleTable {
     fn from(config: &Configuration) -> Self {
         let mut rules = RuleTable::empty();
 
-        let fixable = resolve_codes([RuleCodeSpec {
-            select: config.fixable.as_deref().unwrap_or(&[RuleSelector::All]),
-            ignore: config.unfixable.as_deref().unwrap_or_default(),
-        }]);
+        let mut select_set: FxHashSet<_> = defaults::PREFIXES.iter().flatten().collect();
+        let mut fixable_set: FxHashSet<_> = RuleSelector::All.into_iter().collect();
+
+        let mut redirects = FxHashMap::default();
```

Comment:
Do you mind adding some comments throughout this method? Or explaining the logic in more detail here? I'm tracing it but having trouble following.

As a specific example: if I have `select =["F"]` in my `pyproject.toml`, and I do `--select A`, then I end up with two rule selections:

```rust
[RuleSelection { select: Some([Prefix { prefix: F, redirected_from: None }]), ignore: Some([Prefix { prefix: A, redirected_from: None }]), extend_select: None, extend_ignore: None, fixable: None, unfixable: None }, RuleSelection { select: Some([Prefix { prefix: A, redirected_from: None }]), ignore: None, extend_select: None, extend_ignore: None, fixable: None, unfixable: None }]
```

But, how do the `F` rules not end up in the select set?

Reply:
Oh, I see -- do we, in effect, only use the last `RuleSelection` that contains a non-empty `select`? The rest gets discarded?

Reply:
If that's the correct interpretation how would I do the following: create a `pyproject.toml` that enables all the `F` rules, then extend it from another `pyproject.toml` to also include the `E` rules? I think I know the answer, but it would be helpful to hear it.

Reply:
Separate question: under this scheme, is there _ever_ any difference between specifying `ignore` and `extend-ignore`?

Reply:
I added some comments :)

> if I have select = ["F"] in my pyproject.toml, and I do --select A, [..] But, how do the F rules not end up in the select set?

If `selection.select.is_some()` then the `select_set` is overridden.

> do we, in effect, only use the last RuleSelection that contains a non-empty select? The rest gets discarded?

No, multiple RuleSelections can all contribute to `select_set` if they only use `extend_select` / `ignore` but not `select`.

> how would I do the following: create a pyproject.toml that enables all the F rules, then extend it from another pyproject.toml to also include the E rules?

`select` can be extended via `extend-select`.

> Separate question: under this scheme, is there ever any difference between specifying ignore and extend-ignore?

Very good catch! I didn't even notice that but indeed we can deprecate `extend-ignore` since it now can always be replaced with `ignore`. The reasoning behind that is that now `ignore`/`extend-ignore` can only ever deselect already selected rules ... it's impossible for an ignore option to take precedence over a latter select option. Since the "ignore set" isn't something that is kept track of anymore ... the function only keeps track of the effectively selected rules, `extend-ignore` doesn't fulfill any purpose anymore.

Reply:
Cool, I _think_ this makes sense. I'd like to try adding a section in the README to explain this before we merge. If I do that, do you mind reviewing for clarity and correctness?

Reply:
But of course :)

While I don't think this will affect most ruff configurations this technically still is a breaking change, so we probably should also document it in `BREAKING_CHANGES.md`?

Reply:
Definitely, will do. Mechanically, probably easiest for me to open those changes as a separate PR on `main` and then just rebase that PR after this merges...?

Reply:
You have write permissions on my branch so you could just push your commits there :)

Reply:
Ok, I went ahead and pushed a first-pass in the `README.md` and `BREAKING_CHANGES.md`. Feedback welcome.

Reply:
Thank you so much, I think you described it very well :)

Reply:
Great. I'll try to get this out tomorrow!